src/Mod/PartDesign/Gui/TaskHoleParameters.ui

```diff
@@ -20,30 +20,10 @@
    <string>Task Hole Parameters</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
-   <item row="0" column="0" colspan="2">
-    <widget class="QLabel" name="label_13">
-     <property name="text">
-      <string>&lt;b&gt;Threading and size&lt;/b&gt;</string>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="0">
-    <widget class="QLabel" name="label_2">
-     <property name="sizePolicy">
-      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
-       <horstretch>0</horstretch>
-       <verstretch>0</verstretch>
-      </sizepolicy>
-     </property>
-     <property name="text">
-      <string>Profile</string>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="1" colspan="5">
-    <widget class="QComboBox" name="ThreadType">
+   <item row="19" column="2" colspan="3">
```

Comment:
> Tab order is all screwy due to the random order of the rows and columns.

As I rote you can ignore the order of rows and columns, Qt designer does this automatically it has no effect. And I don't see why we must have them sorted, when Qt's own designer disorders them on every change.

The tab order however, is also incorrect in master. I fixed this now.

Reply:
I prefer if I can see what has changed at a glance by looking at the diff. And the tab order will be decent automatically. Running the script takes a minute, why not do it.

Reply:
> I prefer if I can see what has changed at a glance by looking at the diff.

This is XML code, not designed to be human-readable. So also with sorted column names it won't become readable in mist cases.

> Running the script takes a minute, why not do it.

Because this costs extra time and I would have to do this every time I make a small change in Qt's designer. And it does not lead to the result you like to get. Change for example a grid layout to another layout. The resulting dialog can look almost the same but the diff is horrible. So please ignore the XML diffs. When they would be important, Qt would have designed the designer to maintain human-readable diffs.

Reply:
Terrible argument. The xml is perfectly human readable, by design. The dialog was sorted before so why should it be unsorted now. With the randomized order way you have no control over what has been unintentionally changed in the XML. 
Also, when coding, it is much faster to look in the XML to find out what a widgets name is. It is easy if the items are sorted. The tab order will be decent automatically without having to specify it manually. Lastly, when later you want to find what commit introduced a certain change (git blame). Changing the order every time the dialog is changed will make git blame not work.
These are the advantages of sorting the items. What is the advantages of not sorting?

Reply:
> The dialog was sorted before so why should it be unsorted now.

Again, I am not to blame for design decisions made by Qt. I use their official tool to modify .ui files. Can you please report this to the Qt guys? It cannot be a solution that everyone has to code its own tool to keep the sorting.
And please discuss this in the forum because if we set up the rule to override QT Designer's sorting, every developer must be aware and know how to do this. Please CC there abdullah and wmayer who recently triggered the same discussion.

---------
(Personally, for me XML is in general not readable. To see the widget name, I can just click on the widget in designer and see its name. Opening the XML with a text editor and then searching costs just too much time.)

Reply:
This is not a general issue, it is primarily an issue for the QGridLayout. You are not responsible for Qt designer output of course. You are responsible for the Diff. I have given you the tool. 

I don't mind discussing it in the forum. Would be good to have general guidance. As an example of what I meant with "unintentional changes" to the ui that qt designer  may do. I opened the ui file and saved it. Qt designer 5.14.2 added native="true" and stdset="0" on many elements. Am I not responsible for whatever that change will do to the user experience?  A very strange position to take.

```
-    <widget class="Gui::PrefQuantitySpinBox" name="TaperedAngle">
+    <widget class="Gui::PrefQuantitySpinBox" name="TaperedAngle" native="true">
 ...
-     <property name="minimum">
+     <property name="minimum" stdset="0">
       <double>0.000000000000000</double>
```

We can discuss it further. And perhaps you will have your way. Or you could just apply this change, have a nice diff and clean git history. And we can move on.

Reply:
https://forum.freecadweb.org/viewtopic.php?f=10&t=54503